src/loaders/tvg/tvgTvgLoadParser.cpp

```diff
@@ -0,0 +1,676 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <string.h>
+#include "tvgTvgLoadParser.h"
+
+// Enumeration specifying the loading result
+enum class LoaderResult { InvalidType, Success, SizeCorruption, MemoryCorruption, LogicalCorruption };
+
+// Load paint from .tvg binary file.
+static LoaderResult _parsePaint(tvgBlock block, Paint ** paint);
+
+// Read tvgBlock from pointer location.
+static tvgBlock _readTvgBlock(const char *pointer)
+{
+    tvgBlock block;
+    block.type = *pointer;
+    _read_tvg_ui32(&block.lenght, pointer + TVG_INDICATOR_SIZE);
+    block.data = pointer + TVG_INDICATOR_SIZE + BYTE_COUNTER_SIZE;
+    block.blockEnd = block.data + block.lenght;
+    return block;
+}
+
+/*
+ * Read header of the .tvg binary file
+ * Returns true on success, false otherwise.
+ * Details:
+ * Each file starts with "TVG" phrase and three chars of version number.
+ * Two next bytes are the metadata length (amount of bytes to skip if the meta is not read).
+ * Metadata may contain file generation time and date or other information about device and file.
+ * ["TVG"][version eg. "01a"][uint16 metadata length eg. 0x0000][metadata]
+ *
+ * Starting phase together with metadata are called header.
+ * After header, binary section starts.
+ * NOTE: In whole file little endian is used.
+ */
```

Comment:
As the comments are already made and can help work developing the code further or find difference between saver and loader, I would prefer to leave them

Reply:
Hmm... I tried reading docs. I felt mostly they are useless. Not much helpful. because return values are very descriptive itself. Also duplicated contents with function name + comments among the code.

Reply:
Ok, got your point. Will delete functions comments